src/AspNetCore/Src/forwardinghandler.cxx

```diff
@@ -1447,9 +1447,8 @@ FORWARDING_HANDLER::OnExecuteRequestHandler(
     //
     pResponse->SetStatus(502, "Bad Gateway", 3, hr);
 
-    if (hr > HRESULT_FROM_WIN32(WINHTTP_ERROR_BASE) &&
-        hr <= HRESULT_FROM_WIN32(WINHTTP_ERROR_LAST))
-    {
+    if (!(hr > HRESULT_FROM_WIN32(WINHTTP_ERROR_BASE) &&
+          hr <= HRESULT_FROM_WIN32(WINHTTP_ERROR_LAST)) ||
```

Comment:
The code logic was change here. In the old code if FormatMessage failed, we will not try LoadString. 
This change make sense to me.